tools/go-ls-tags/files/reader.go

```diff
@@ -0,0 +1,54 @@
+/*
+Copyright 2022 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package files
+
+import (
+	"bufio"
+	"context"
+	"os"
+
+	"knative.dev/test-infra/tools/go-ls-tags/timeout"
+)
+
+// ReadLines will read lines from a file and pass each line to a func handler.
+func ReadLines(ctx context.Context, path string, fn func(line string) error) error {
+	f, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+	defer func(f *os.File) {
```

Comment:
Interesting, why isn't it just `defer f.Close()` ?

Reply:
To silence the linter in IDE.

With `defer f.Close()` the IDE complains that return value from `Close` isn't utilized. This way, it should be obvious, that omitting the return value is intentional.

The `f.Close() could return error in some cases. For example, you are writing to a disk that is rapidly filling. The contents of the write are small, so it gets buffered, and `f.Flush()` is called within `.Close()`. Then the error would be returned.

This file is used just for reading, so I believe there is no way an error could be returned.

Reply:
Good to know. But it was not clear to me at first that this would be there to silence the linter. I know that Close() can return an error and calling defer with it directly makes it ignore the error. Personally, I'd use this to make it clear:
 ```
defer f.Close() //nolint // Ignore returned error
```

Reply:
Or maybe just a comment would be helpful. Thanks